src/Cache/_ScsDataClient.php

```diff
@@ -354,14 +369,92 @@ public function listErase(string $cacheName, string $listName, ?int $beginIndex=
                 $listEraseRequest->setAll($all);
             }
             $call = $this->grpcManager->client->ListErase(
-                $listEraseRequest, ["cache"=>[$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]
+                $listEraseRequest, ["cache" => [$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]
             );
             $this->processCall($call);
         } catch (SdkError $e) {
             return new CacheListEraseResponseError($e);
-        } catch (\Exception $e) {
+        } catch (Exception $e) {
             return new CacheListEraseResponseError(new UnknownError($e->getMessage()));
         }
         return new CacheListEraseResponseSuccess();
     }
+
+    public function dictionarySet(string $cacheName, string $dictionaryName, string $field, string $value, bool $refreshTtl, ?int $ttlSeconds = null): CacheDictionarySetResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateDictionaryName($dictionaryName);
+            validateFieldName($field);
+            validateValueName($value);
+            $ttlMillis = $this->ttlToMillis($ttlSeconds);
+            $dictionarySetRequest = new _DictionarySetRequest();
+            $dictionarySetRequest->setDictionaryName($dictionaryName);
+            $dictionarySetRequest->setItems([$this->toSingletonFieldValuePair($field, $value)]);
+            $dictionarySetRequest->setRefreshTtl($refreshTtl);
+            $dictionarySetRequest->setTtlMilliseconds($ttlMillis);
+            $call = $this->grpcManager->client->DictionarySet($dictionarySetRequest, ["cache" => [$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]);
```

Comment:
I recommend pre-computing this so we don't multiply this out in all the functions, ie in the constructor store `deadline_microseconds = $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER` and use that here

Reply:
I'd rather avoid using `deadline_microseconds` because they're technically not microseconds. I think this is a great idea, but I'd like to think about it a bit . . . I'll make this change in an upcoming PR.